pulsar-io/redis/src/main/java/org/apache/pulsar/io/redis/sink/RedisSink.java

```diff
@@ -121,9 +121,9 @@ private void flush() {
         if (CollectionUtils.isNotEmpty(recordsToFlush)) {
             for (Record<byte[]> record: recordsToFlush) {
                 try {
-                    // records with null keys or values will be ignored
-                    byte[] key = record.getKey().isPresent() ? record.getKey().get().getBytes(StandardCharsets.UTF_8) : null;
+                    // use the value of record as key when its key is null
                     byte[] value = record.getValue();
+                    byte[] key = record.getKey().isPresent() ? record.getKey().get().getBytes(StandardCharsets.UTF_8) : value;
```

Comment:
why are you using the `value` as key ?

probably if the record has no "key" we should discard it 
mapping a null key to the value is not immediate for the user

Reply:
A null key cannot flush to redis, `java.lang.NullPointerException: null` may confuse someone who will treat as no record is coming.

Reply:
Maybe an empty string `""` looks better?

Reply:
Empty String makes more sense to me.

Reply:
Done, please take a look again, thanks.